rest/OrderService.js

```diff
@@ -2,35 +2,66 @@ import ProductPersistence from "../persistence/ProductPersistence.js";
 import OrderPersistence from "../persistence/OrderPersistence.js";
 import IllegalArgumentError from "./IllegalArgumentError.js";
 
+const EMPTY_ARGUMENT_ERR = "Body contains an empty argument!";
+const INVALID_TIMESTAMP_ERR = "Timestamp create should be now!";
+
 class OrderService {
-    #productPersistence;
     #orderPersistence;
 
     constructor() {
-        this.#productPersistence = new ProductPersistence();
         this.#orderPersistence = new OrderPersistence();
     }
 
     async createOrder(order) {
-        return await this.#orderPersistence.create(order);
+        this.#assertNoEmptyArgs(order);
+        this.#assertValidTimestamp(order.timestampCreated);
+        return await this.#orderPersistence.create(
```

Comment:
Also maybe check for isAutomated, deleting automated orders will probably break someone's automation

Reply:
Actually do we check if the product even exists? The exists check is only in the persistence and validates deletion, updates and fetching. We don't want to use that since the client will get 404 if they input an invalid product, we want them to get 400.